tls/s2n_config.c

```diff
@@ -427,9 +459,37 @@ int s2n_config_add_cert_chain_and_key_with_status(struct s2n_config *config, con
 
     config->cert_and_key_pairs->chain_size = chain_size;
 
-    if (status && length > 0) {
-        GUARD(s2n_alloc(&config->cert_and_key_pairs->ocsp_status, length));
-        memcpy_check(config->cert_and_key_pairs->ocsp_status.data, status, length);
+    for (uint16_t ext_idx = 0; ext_idx < num_extensions; ext_idx++) {
+        switch (extensions[ext_idx].type) {
+            case S2N_EXTENSION_CERTIFICATE_TRANSPARENCY: {
```

Comment:
Both of the cases in the switch statements can be simplified so as to avoid so much indentation. 

```
case S2N_EXTENSION_CERTIFICATE_TRANSPARENCY: {
    if (have_sct || extensions[ext_idx].data == NULL || extensions[ext_idx].length == 0) {
        S2N_ERROR(S2N_ERR_INVALID_SCT_LIST);
    } 
    have_sct = 1;
    GUARD(s2n_alloc(&config->cert_and_key_pairs->sct_list, extensions[ext_idx].length));
    memcpy_check(config->cert_and_key_pairs->sct_list.data, extensions[ext_idx].data, extensions[ext_idx].length);
} break;
```